### PR TITLE
image_generation file

### PR DESCRIPTION
**Pull request by mapbobi-commit**

### CODE REVIEW

File: src/image_generation.py
Comment (maxvelkir) [inline src/image_generation.py:1]: Put some logical separation between the lines of code with blank lines. Everything is glued together and it's hard to follow the code.

File: src/image_generation.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    response = openai.Image.create(
+        prompt=prompt,
+        n=n,
+        size=size,
+        response_format=response_format,
+    )
+    images = [v[response_format] for v in response["data"]]
+    images = [b64decode(v) for v in images]
+    images = [io.BytesIO(v) for v in images]
+    images = [Image.open(v) for v in images]
```
Comment (maxvelkir) [inline src/image_generation.py:41]: This is a clean look but it's O(n^4) complexity to complete which is crazy bad

File: src/image_generation.py
Comment (maxvelkir) [inline src/image_generation.py:1]: I also want to be able to generate the product first with a description and only then to generate the images

**Review by maxvelkir — CHANGES_REQUESTED** [review]

File: src/image_generation.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+os.environ["OPENAI_API_KEY"] = config["auth"]["api_key"]
+
+llm_prompt = "Here is a csv table describing products.\n{table_str}\n. Based on those existing products, generate a novel product providing a short textual description of the product."
+img_prompt = "Create realistic waist down photos of a model wearing following:{}"
+dataframe_csv_table=pd.read_csv("../data/ex1/table.csv")
+
+def make_llm_chain(temp, template):
+    llm = OpenAI(temperature=temp)
+    prompt = PromptTemplate(template=template, input_variables=["table_str"])
+    chain = LLMChain(llm=llm, prompt=prompt)
```
[the remark this answers is not in the document]
Comment (mapbobi-commit) [reply src/image_generation.py:21]: In their documentation, it says that langchain standardizes prompts and llm chains. If we ever go to a different llm (AI), we could still use langchain.
